src/Model/FilterJob.re

```diff
@@ -135,35 +136,28 @@ module Make = (Config: Config) => {
     (false, newPendingWork, completed);
   };
 
+  /* Compare two ranked items. */
+  let compareItems = (a, b) => compare(a.Filter.score, b.Filter.score);
+
   let doActualWork =
       (
         {queue, shouldLower, filter, explodedFilter, _} as pendingWork: PendingWork.t,
-        {filtered, _}: CompletedWork.t,
+        {ranked, _}: CompletedWork.t,
       ) => {
     // Take out the items to process this frame
     let (items, queue) = Queue.take(Constants.itemsPerFrame, queue);
 
-    // Do a first filter pass to check if the item satisifies the regex
-    let filtered =
-      List.fold_left(
-        (acc, item) => {
-          let name = format(item, ~shouldLower);
-
-          if (Filter.fuzzyMatches(explodedFilter, name)) {
-            [item, ...acc];
-          } else {
-            acc;
-          };
-        },
-        filtered,
-        items,
-      );
+    // TODO: Understand why this is needed.
+    let filtered = items;
 
     // Rank a limited nuumber of filtered items
+    // TODO: Should the mergeSortedList include a limit? Faster to quick return
+    // rather than merge then stop.
     let ranked =
-      filtered
-      |> ListEx.firstk(Constants.maxItemsToFilter)
-      |> Filter.rank(filter, format);
+      items
+      |> Filter.rank(filter, format)
+      |> ListEx.mergeSortedList(compareItems, ranked)
+      |> ListEx.firstk(Constants.maxItemsToFilter);
```

Comment:
I probably don't want this, since I'm doing extra work for no reason. Should integrate early stopping into the merging.

Reply:
That sounds like a good idea!